ckanext/harvest/harvesters/ckanharvester.py

```diff
@@ -528,6 +529,33 @@ def get_extra(key, package_dict):
                 # key.
                 resource.pop('revision_id', None)
 
+
+            # validate packages if needed
+            validate_packages = self.config.get('validate_packages', {})
+            if validate_packages:
+                if 'type' not in package_dict:
+                    package_plugin = lib_plugins.lookup_package_plugin()
+                    try:
+                        # use first type as default if user didn't provide type
+                        package_type = package_plugin.package_types()[0]
+                    except (AttributeError, IndexError):
+                        package_type = 'dataset'
+                        # in case a 'dataset' plugin was registered w/o fallback
+                        package_plugin = lib_plugins.lookup_package_plugin(package_type)
```

Comment:
Shouldn't that line be one level higher? Otherwise `package_plugin` is not set if there was no exception on line 540.

Reply:
package_plugin is set on line 537 and most of that code is actually copy paste from here https://github.com/ckan/ckan/blob/master/ckan/logic/action/create.py#L136

Reply:
Ah perfect, thanks for the explanation!